### PR TITLE
Macro examples added to Release Notes template

### PR DESCRIPTION
**Pull request by PeterTurcan**
Macro examples taken from existing sample Release Notes.

### CODE REVIEW

**Review by alandefreitas — COMMENTED** [review]
Where is this file going to be used? If it's not rendered with Antora, then we can't use our macros either (we're not using them yet, but just making the point for when we fix this `phrase library` thing.

File: release-notes/template.adoc [hunk 1]
```diff
@@ -1,59 +1,121 @@
...
-// List new libraries in lexicographical order (library name, purpose, authors)
+
+// List new libraries in lexicographical order:
 // Example:
-// * Accumulators : performs incremental calculation, and collection of statistical accumulators, from Eric Niebler.
-* Library 1 name : purpose, author
-* Library 2 name : purpose, author
+// * [phrase library..[@/libs/accumulators/ Accumulators]:] : performs incremental calculation, and collection of statistical accumulators, from Eric Niebler.
+
+* [phrase library..[@/libs/charconv/ Charconv]:] A high quality implementation of `<charconv>` in C++11, from Matt Borland.
```
[the remark this answers is not in the document]
Comment (alandefreitas) [reply release-notes/template.adoc:21]: If I understand correctly, @joaquintides wanted an adoc template of how adoc release notes should be. Not an adoc template explaining how qbk release notes should be. I might be wrong about that though.

**Comment by joaquintides** [comment]
Sorry for the confusion, I specified this very poorly. The current qbk-based mechanism makes available some "macros" defined here:

https://github.com/boostorg/website/blob/master/feed/ext.qbk

Many of these macros are not in actual use in the release notes (meaning I haven't seen them used recently or ever). The ones that I've seen are (see the latest [qbk-based release notes](https://github.com/boostorg/website/blob/master/feed/history/boost_1_85_0.qbk) and the [generated output](https://www.boost.org/users/history/version_1_85_0.html)):

* `[phrase library..XX]`: produces a header for the section on lib X.
* `[@/libs/function/ Function]` creates a link to the docs for the library (Function in the example). I understand that 
* `[github_pr geometry 1247]`: produces this: [PR#1247](https://github.com/boostorg/geometry/pull/1247)
* `[github geometry 1231]`: produces this: [#1231](https://github.com/boostorg/geometry/issues/1231)
* `[@/tools/build/ Build]`: I think this is formally equivalent to `[@/libs/function/ Function]`.

Now, a question: do we have the ability to create macros like these for our new Asciidoc-based release notes?

* If so, we should that for at least the macros listed above and include examples uf use in the template.
* If not, then we should at least show how the equivalent macro-less expressions should be written.

Happy to discuss things further and/or have a videocall if this would help.

Thank you!

**Comment by PeterTurcan** [comment]
Closing this for now. Will re-address after decisions made on support of macros in adoc-based release notes.

**Pull request closed without merging**